### ISSUE TITLE
while setting variables

### ISSUE THREAD

**Issue by palashmoon** [cosmaadrian/multimodal-depression-from-video#85]
Hello, can you please help me with what value I should set for the variable in each file ?? for example PASE_CONFIG=
PASE_CHCK_PATH=
USE_AUTH_TOKEN= what should be the value for this??

**Comment by david-gimeno** [comment]
Hi Palash,

I guess we are talking about the script [./scripts/feature_extraction/extract-dvlog-pase+-feats.sh](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/scripts/feature_extraction/extract-dvlog-pase%2B-feats.sh) for extracting the audio-based PASE+ features. Different aspects to take into account:

· The videos of the dataset are expected to be placed at ``./data/D-vlog/videos/'', as indicated by variable $VIDEO_DIR.
· You should clone the official PASE+ repo: https://github.com/santi-pdp/pase.git
· You then have to follow the instructions [here](https://github.com/santi-pdp/pase?tab=readme-ov-file#pre-trained-model) to download the pre-trained model provided by the original authors and to know the config file you should use.

Therefore, once you have cloned the repo and downloaded the pre-trained model checkpoint, you should be able to set these variables, e.g., as follows:

PASE_CONFIG=./pase/cfg/frontend/PASE+.cfg
PASE_CHCK_PATH=./pase/FE_e199.ckpt

Regarding the USE_AUTH_TOKEN variable, is a commonly required authentication token when using certain HuggingFace models. Please find the instructions to use PyAnnote [here](https://huggingface.co/pyannote/voice-activity-detection).

**Comment by palashmoon** [comment]
Thank you @david-gimeno for the help. 
can you help me in setting up these variables too? 
1. INSTBLINK_CONFIG=
2. ETH_XGAZE_CONFIG=
3. BODY_LANDMARKER_PATH= ./data/D-vlog/body_landmarks
HAND_LANDMARKER_PATH= ./data/D-vlog/hand_landmarks.

**Comment by david-gimeno** [comment]
You can find the toolkit we employed to extract each modality in the [paper](https://arxiv.org/pdf/2401.02746) (see page no. 7). However, I am seeing that is not as easy as I expected configuring all these feature extractors. Let's go step by step.

- **Configuring InstBlink:** You should clone the following [repo](https://github.com/wenzhengzeng/MPEblink) and download the checkpoint indicated [here](https://github.com/wenzhengzeng/MPEblink?tab=readme-ov-file#inference--evaluation). Then, if you inspect the [script](https://github.com/wenzhengzeng/MPEblink/blob/main/tools/test_eval.sh) the original authors wrote, we can infer the following configuration: 
              `INSTBLINK_CONFIG=./MPEblink/configs/instblink/instblink_r50.py`
              `INSTBLINK_CHCK_PATH=./MPEblink/pretrained_models/instblink_r50.pth`

- **Configuring ETH_XGaze:** Similarly, you should clone the following [repo](https://github.com/hysts/pytorch_mpiigaze_demo). The authors provide config files for various models. In our work, we used the ETH-XGaze detector. So, the config file should be as follows:
              `ETH_XGAZE_CONFIG=./pytorch_mpiigaze_demo/ptgaze/data/configs/eth-xgaze.yaml`

- **Configuring MediaPipe's Models:** As indicated in the paper, we based our body and hand landmarkers on MediaPipe by Google. Take into account that this platform is plenty of models and you are not limited to using the same as ours. However, I can share the specific models we employed. If you read this [tutorial](https://colab.research.google.com/github/googlesamples/mediapipe/blob/main/examples/pose_landmarker/python/%5BMediaPipe_Python_Tasks%5D_Pose_Landmarker.ipynb?hl=es-419#scrollTo=OMjuVQiDYJKF) you can find the Linux command to download the model checkpoint for the body pose estimator. For the hand landmark detector, you can find the checkpoint downloading Linux command [here](https://colab.research.google.com/github/googlesamples/mediapipe/blob/main/examples/hand_landmarker/python/hand_landmarker.ipynb?hl=es-419#scrollTo=OMjuVQiDYJKF). How did I find this tutorial? Search in google for `hand landmark mediapipe` and then click on  `Python Code Example`. Therefore, once you download the model checkpoints, the configuration should be something as follows:

              `BODY_LANDMARKER_PATH=./landmarkers/pose_landmarker.task`
              `HAND_LANDMARKER_PATH=./landmarkers/hand_landmarker.task`

_Note that the `/landmarkers/` directory is not automatically created, it was just not do a mess in our code_

**Comment by palashmoon** [comment]
Thank you @david-gimeno for the quick response.
I cloned the repo as you mentioned in the above comment for gaze and used this >ETH_XGAZE_CONFIG=./pytorch_mpiigaze_demo/ptgaze/data/configs/eth-xgaze.yaml
but using this i giving me one more issue which is 
![image](https://github.com/cosmaadrian/multimodal-depression-from-video/assets/14892577/d03c7d7e-6940-4d5e-81eb-4876c6cddb96)
one more thing why are cloning pytorch_mpiigaze_demo.. and how can i resolve this issue??
Thank you so much once again

**Comment by david-gimeno** [comment]
I would need more details. What OS are you using, Ubuntu? and How did you set the variable `MPIIGAZE_DIR=`?

**Comment by palashmoon** [comment]
Hi, I am using Ubuntu only I have set like this. 

MPIIGAZE_DIR=./scripts/conda_envs/feature_extractors/pytorch_mpiigaze_demo/
ETH_XGAZE_CONFIG=./scripts/conda_envs/feature_extractors/pytorch_mpiigaze_demo/ptgaze/data/configs/eth-xgaze.yaml

please let me know if you need any other information.

**Comment by david-gimeno** [comment]
According to the [script](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/scripts/feature_extraction/dvlog/extract_gaze_tracking.py#L90) the model checkpoints should automatically be downloaded. So, let's try using absolute paths, just in case.

**Comment by david-gimeno** [comment]
Okey, I think the problem should be in the config file `pytorch_mpiigaze_demo/ptgaze/data/configs/eth-xgaze.yaml`. Open it with a text editor and modify the paths to the checkpoints according to the way you structure your project.

**Comment by palashmoon** [comment]
Hi @david-gimeno . I have updated the file like this 
gaze_estimator:
  **checkpoint: ./scripts/conda_envs/feature_extractors/pytorch_mpiigaze_demo/ptgaze/models/eth-xgaze_resnet18.pth**
  camera_params: ${PACKAGE_ROOT}/data/calib/sample_params.yaml
  use_dummy_camera_params: false
  normalized_camera_params: ${PACKAGE_ROOT}/data/normalized_camera_params/eth-xgaze.yaml
  normalized_camera_distance: 0.6
  image_size: [224, 224]
but still I am getting the same error.

**Comment by david-gimeno** [comment]
Please try with an absolute path: `/home/....` If the error persists you should contact the original authors of the gaze tracker code.

**Comment by palashmoon** [comment]
Thank you @david-gimeno i was able to extract gaze feature.. after downloading the model it was getting stored in the path as 

> ~/.patze./... 
which was not able to expand properly by python I use os.expanduser(~/.patze/..) to get the correct path.
Thanks for the help again.

**Comment by palashmoon** [comment]
Hi @david-gimeno. I am trying to extract the emonet features using the following code it is looking for a face_ID inside the directory. 
![image](https://github.com/cosmaadrian/multimodal-depression-from-video/assets/14892577/33bf5186-f3c0-4d27-b1a8-1657a29bd695)
But the current structure is like this. 
![image](https://github.com/cosmaadrian/multimodal-depression-from-video/assets/14892577/fadf438f-ec0c-4aea-b877-9f290458e7c5)

following is the error I am getting. 
![image](https://github.com/cosmaadrian/multimodal-depression-from-video/assets/14892577/f1097f23-f64f-4496-a403-5d2175f87a12)
can you please provide some suggestion on this??

**Comment by david-gimeno** [comment]
It seems the script was expecting an additional level of directories. So, the script has been modified, you can check it [here](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/scripts/feature_extraction/dvlog/extract_face_emonet_features.py). Please update the repo and try again.

Regarding the requirements for each feature extractor, we provide info in the README of the repo. Please, read it carefully. Nonetheless, take into account that these installations usually depend on the OS architecture and might fail on certain occasions. Issues related to these installations should be solved by contacting the original authors of the corresponding models.

**Comment by palashmoon** [comment]
@david-gimeno is there any workaround for the installation of requirement.txt.. there are few installation issues I am facing. i have raised issue with the original authors too. 
https://github.com/santi-pdp/pase/issues/128. these are current errors I am getting while installation.

**Comment by palashmoon** [comment]
HI @david-gimeno.. I have downloaded all the videos from the D-vlog dataset.. should i split it based on the ids given in the test, train and validation.csv file?? or there is a separate file **python3 ./scripts/feature_extraction/dvlog/extract_wavs.py --csv-path ./data/D-vlog/video_ids.csv --column-video-id video_id --video-dir $VIDEO_DIR --dest-dir $WAV_DIR** video_id.csv which is missing in the repo?? please help me with this..

**Comment by waHAHJIAHAO** [comment]
Hi @palashmoon ，could you share your d-vlog dataset，I have  been looking for this for a long time.

**Comment by bucuram** [comment]
@waHAHJIAHAO please reach out to the authors of the D-vlog dataset to request access: [D-vlog: Multimodal Vlog Dataset for Depression Detection](https://ojs.aaai.org/index.php/AAAI/article/view/21483)

**Comment by waHAHJIAHAO** [comment]
Hi,  @bucuram ,May I ask how to fill this config file.
<img width="271" alt="image" src="https://github.com/user-attachments/assets/5516c29e-a711-40fd-a32f-c51ca1f40f6c">

**Comment by bucuram** [comment]
Hi, @waHAHJIAHAO!

There should be the paths to the data, for example:

```
reading-between-the-frames:
  d-vlog: data/D-vlog/
  d-vlog-original: data/D-vlog/splits/original/
  daic-woz: data/DAIC-WOZ/
  e-daic-woz: data/E-DAIC-WOZ/
  num_workers: 8
```

**Comment by waHAHJIAHAO** [comment]
Thanks @bucuram  a lot!! and What should I fill in the field of "ewing-between-the-frames"

**Comment by bucuram** [comment]
That field should remain as it is in the example above. Then, you can use the env config when running the experiments.

The env config is already set as `ENV="reading-between-the-frames"`.
https://github.com/cosmaadrian/multimodal-depression-from-video/blob/d19568cb43ebbbfaf43de02a2cfdbe5fc621bc6a/experiments/original-dvlog/train-baseline-original-dvlog-modality-ablation.sh#L20

**Comment by waHAHJIAHAO** [comment]
@bucuram YESS!!，thank u for reply~~ ,due to I want to process new dataset which collected from my university,I have some quetions about d-vlog dataset . the filepath is "data/D-vlog/splits",which contain 4 csv files."voice_presence	face_presence	body_presence	hand_presence"
These four fields were originally present in the D-VLOG dataset, or were they processed by your team later. If you did the pre-processing operation later, please tell me how to generate this part of data. And does the absence of this part have any effect on the model?
<img width="1171" alt="image" src="https://github.com/user-attachments/assets/39657516-38d0-4632-a86f-0843f821ecb0">
this is part of my dataset:
<img width="1022" alt="image" src="https://github.com/user-attachments/assets/09d25059-ff80-460f-bba2-16da514fe9c5">

**Comment by david-gimeno** [comment]
Hi @waHAHJIAHAO!

These four dataframe columns **were not** originally in the D-Vlog dataset. We computed these statistics thanks to our feature extraction scripts you can find [in this directory](https://github.com/cosmaadrian/multimodal-depression-from-video/tree/master/scripts/feature_extraction/dvlog). As you can observe, for example, in the [face detection script](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/scripts/feature_extraction/dvlog/extract_faces_and_landmarks.py#L38) and [body landmarks identification script](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/scripts/feature_extraction/dvlog/extract_body_pose_landmarks.py#L56), we were creating a numpy array with the index of those frames were no face or no body was detected. Additionally, as you can also notice, we were zero-filling the final feature sequence representing the video sample.

So, how did we compute that voice, face, etc. presence values? Having the information I mentioned above and knowing the frame per seconds of each video clip, we can compute the number of seconds were the subject was actually talking, actually present in the scene, etc. These are statistics to know more things about the dataset. What we actually used for model training was the array with the index where there was, e.g., no face, to create a mask with 0's and 1's to tell the model where it shouldn't pay attention.

**Comment by waHAHJIAHAO** [comment]
Hi @david-gimeno ~，Thank you for your careful reply. I have completed the pre-processing of the "presence" part. I still pre-processed my new data set according to D-vlog, generated some segmented npz files to access the model, and wrote the following data processing scripts which are logically consistent with dvlog. The current error problem is that my "traindataloader" did "torch.stack()" loading data error. I read on the fifth page of your paper that there is a learnable modal encoder that can unify the output of each mode. I would like to ask where is this encoder, or do you have any suggestions for my problem?
<img width="281" alt="image" src="https://github.com/user-attachments/assets/f8492c45-2e10-4375-9b73-8968da6c1b19">
<img width="547" alt="image" src="https://github.com/user-attachments/assets/aebeeaf0-07ef-44d8-af29-3c9e2afb45d9">
this is my error:
<img width="759" alt="image" src="https://github.com/user-attachments/assets/09c77f53-6126-45ce-8da2-f133b0a861fd">

**Comment by waHAHJIAHAO** [comment]
I just use 5 modality and print their shape ，it look like this：
<img width="374" alt="image" src="https://github.com/user-attachments/assets/e5df98bc-3f56-460d-b0ac-39a85132b23b">

**Comment by david-gimeno** [comment]
@waHAHJIAHAO 
The tensor shapes look nice. I guess 270 refers to the number of frames composing your context window, which, if your videos were recorded at 25fps, should correspond to a 10-second span. Are you using our code or are you implementing your own dataset? Anyway, I recommend you to carefully inspect our dataset script, specifically [here](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/datasets/dvlog_dataset.py#L122) you have a good starting point. You can debug how you data shape looks at every dataset step, either some tools or simple, yet effective, `print()` and `exit()`, 

Regarding the learnable modality encoders to unify all inputs to the same dimensional space, you can find their implementation [here](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/models/modality_encoders.py). These modality encoders are subsequently used [here](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/models/baseline_model.py#L52) when defining our Transformer-based model. Note that some of the modalities will be flatten (check this [code](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/models/modality_encoders.py#L99) and this [config file](https://github.com/cosmaadrian/multimodal-depression-from-video/blob/master/configs/modality_config.yaml#L33)). I agree our model do and takes into account a lot of details, but I believe that going step by step to understand the code is the proper way and it will not be for sure a waste of time :)

**Comment by waHAHJIAHAO** [comment]
@david-gimeno Thank u for reply！！！！I have been run my dataset successfully，but i found a question：It does not seem to converge when training, the best results will occur in a few epochs, and 200 epochs seem unnecessary
